diagram.schema.json

```diff
@@ -0,0 +1,240 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Diagram",
+  "type": "object",
+  "required": [
+    "ops"
+  ],
+  "properties": {
+    "ops": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/DiagramOperation"
+      }
+    }
+  },
+  "definitions": {
+    "DiagramOperation": {
+      "oneOf": [
+        {
+          "description": "Signifies the start of a workflow. There must be exactly 1 start operation in a diagram.",
+          "type": "object",
+          "required": [
+            "next",
+            "type"
+          ],
+          "properties": {
+            "next": {
+              "type": "string"
+            },
+            "type": {
+              "type": "string",
+              "enum": [
+                "start"
+              ]
+            }
+          }
+        },
+        {
+          "description": "Signifies the end of a workflow. There must be exactly 1 terminate operation in a diagram.",
+          "type": "object",
+          "required": [
+            "type"
+          ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": [
+                "terminate"
+              ]
+            }
+          }
+        },
+        {
+          "description": "Connects the request to a registered node.\n\n``` # bevy_impulse::Diagram::from_json_str(r#\" { \"ops\": { \"start\": { \"type\": \"start\", \"next\": \"nodeOp\" }, \"nodeOp\": { \"type\": \"node\", \"nodeId\": \"myNode\", \"next\": \"terminate\" }, \"terminate\": { \"type\": \"terminate\" } } } # \"#)?; # Ok::<_, serde_json::Error>(())",
+          "type": "object",
+          "required": [
+            "next",
+            "nodeId",
```

Comment:
I'm a bit reluctant to call this a `nodeId` because
1. Users may think this is supposed to be a UUID rather than an idiomatic name
2. Users may assume every node needs a unique `nodeId`
3. This ID refers to a node **_builder_**, not a node
4. Users may get confused between the purpose of this field and the _name_ of the node (the node's key in the `ops` dictionary)

I would consider using another name for this like `provider` or `builder`.

Reply:
ca569172a3e2b1214625b53046be7d95621803ef

Reply:
Nice, we'll just need to update the [`NodeNotFound`](https://github.com/open-rmf/bevy_impulse/commit/ca569172a3e2b1214625b53046be7d95621803ef#diff-e7190dc1175e279538fa4f9d17a6cbb46119d50677ed0eeea978f26ca17a119bR402) variant to `BuilderNotFound`.

Reply:
01d3bb61e3a429374caf2ca9ebde931932c14c71